core/domain/summary_services.py

```diff
@@ -163,6 +163,90 @@ def get_displayable_collection_summary_dicts_matching_ids(collection_ids):
     return _get_displayable_collection_summary_dicts(collection_summaries)
 
 
+def get_collection_node_metadata_dicts(query_string, search_cursor, user_id):
```

Comment:
This function has nothing to do with collection nodes. Maybe call it get_exp_metadata_dicts_matching_query().

Also, shouldn't it have at least one test?